pkg/controller/controller_instance.go

```diff
@@ -964,21 +964,26 @@ func (c *controller) pollServiceInstance(instance *v1beta1.ServiceInstance) erro
 			return c.finishPollingServiceInstance(instance)
 		}
 
-		// We got some kind of error and should continue polling.
-		//
-		// The instance's Ready condition should already be False, so
-		// we just need to record an event.
 		reason := errorPollingLastOperationReason
 		message := fmt.Sprintf("Error polling last operation: %v", err)
-		glog.V(4).Info(pcb.Message(message))
-		c.recorder.Event(instance, corev1.EventTypeWarning, reason, message)
+		readyCond := newServiceInstanceReadyCondition(v1beta1.ConditionFalse, reason, message)
 
 		if c.reconciliationRetryDurationExceeded(instance.Status.OperationStartTime) {
-			readyCond := newServiceInstanceReadyCondition(v1beta1.ConditionFalse, reason, message)
 			return c.processServiceInstancePollingFailureRetryTimeout(instance, readyCond)
 		}
 
-		return c.continuePollingServiceInstance(instance)
+		if httpErr, ok := osb.IsHTTPError(err); ok {
+			if isRetriableHTTPStatus(httpErr.StatusCode) {
```

Comment:
I was on the fence whether to keep retrying on getting `400 BadRequest`.
It doesn't make sense to retry polling until the issue is fixed on the broker side though.

It seems okay to do `svcat touch instance ...` after fixing the issue on the broker side?

Reply:
I agree that a 400 is a terminal error that should stop polling so the user can fix and then use touch to retry try it. We are working on a new command `svcat set instance --param foo=bar` that will let you fix an instance and touch it in a single command, but it's not merged yet.